src/Map/src/MapOptionsNormalizer.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+use Symfony\UX\Map\Bridge as MapBridge;
+use Symfony\UX\Map\Exception\UnableToDenormalizeOptionsException;
+use Symfony\UX\Map\Exception\UnableToNormalizeOptionsException;
+
+/**
+ * Normalizes and denormalizes map options.
+ *
+ * @internal
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final class MapOptionsNormalizer
+{
+    private const string KEY_PROVIDER = '@provider';
+
+    /**
+     * @var array<string, class-string<MapOptionsInterface>>
+     */
+    public static array $providers = [
+        'google' => MapBridge\Google\GoogleOptions::class,
+        'leaflet' => MapBridge\Leaflet\LeafletOptions::class,
+    ];
```

Comment:
What about a constant ?

Reply:
https://3v4l.org/Sk47Z#v8.3.14

If this array is readonly, let's use a const no ?

Reply:
Ah! I thought you were suggesting using constants for the keys on the board.

Well, the array is supposed to be readonly yes, **but** it is modified in tests to register the `\Symfony\UX\Map\Tests\DummyOptions` class, and reflection does not help here.

Reply:
Well then... it's a good sign it's way to early for micro optimization :)